lib/manager/maven/extract.js

```diff
@@ -0,0 +1,117 @@
+const xmlParser = require('fast-xml-parser');
+const { isVersion } = require('../../versioning/maven');
+
+function parsePom(raw) {
+  if (!xmlParser.validate(raw)) return null;
+  const xml = xmlParser.parse(raw, {
+    ignoreAttributes: false,
+    parseTrueNumberOnly: false,
+  });
+  if (!xml) return false;
+  const project = xml.project;
+  if (!project) return false;
+  const ns = project['@_xmlns'];
+  if (!ns) return false;
+  if (ns !== 'http://maven.apache.org/POM/4.0.0') return false;
+  return project;
+}
+
+function containsPlaceholder(str) {
+  return /\${.*?}/g.test(str);
+}
+
+function extractFromNode(node, mavenXmlPath) {
+  if (node.version) {
+    const groupId = node.groupId || null;
+    const artifactId = node.artifactId || null;
+    const depName = `${groupId}:${artifactId}`;
+    const currentValue = node.version ? node.version.toString() : null;
+    const result = {
+      depName,
+      currentValue,
+      mavenXmlPath,
+    };
+    if (!groupId || !artifactId) {
+      result.skipReason = 'name-incomplete';
```

Comment:
Is this normal? or a strange error?

Reply:
Just to ensure we have both fields so `depName` is valid (i.e. not `null:artifact-id`)

Reply:
Yes good to add safety checks but it would probably be an invalid Pom file if so?

Reply:
Well, I thought noticed example when one of them was absent, but can't find it now. Anyway you're right, I need to get rid of this and check the  a bit earliear.